src/adapter/src/coord/timestamp_oracle/postgres_oracle.rs

```diff
@@ -256,6 +255,134 @@ impl PostgresClientKnobs for PostgresTimestampOracleConfig {
+    pub pg_connection_pool_max_size: Option<usize>,
+    /// Configures [`DynamicConfig::pg_connection_pool_max_wait`].
+    pub pg_connection_pool_max_wait: Option<Duration>,
+    /// Configures [`DynamicConfig::pg_connection_pool_ttl`].
+    pub pg_connection_pool_ttl: Option<Duration>,
+    /// Configures [`DynamicConfig::pg_connection_pool_ttl_stagger`].
+    pub pg_connection_pool_ttl_stagger: Option<Duration>,
+    /// Configures [`DynamicConfig::pg_connection_pool_connect_timeout`].
+    pub pg_connection_pool_connect_timeout: Option<Duration>,
+    /// Configures [`DynamicConfig::pg_connection_pool_tcp_user_timeout`].
+    pub pg_connection_pool_tcp_user_timeout: Option<Duration>,
+}
+
+impl PostgresTimestampOracleParameters {
+    /// Update the parameter values with the set ones from `other`.
+    pub fn update(&mut self, other: PostgresTimestampOracleParameters) {
+        // Deconstruct self and other so we get a compile failure if new fields
+        // are added.
+        let Self {
+            pg_connection_pool_max_size: self_pg_connection_pool_max_size,
+            pg_connection_pool_max_wait: self_pg_connection_pool_max_wait,
+            pg_connection_pool_ttl: self_pg_connection_pool_ttl,
+            pg_connection_pool_ttl_stagger: self_pg_connection_pool_ttl_stagger,
+            pg_connection_pool_connect_timeout: self_pg_connection_pool_connect_timeout,
+            pg_connection_pool_tcp_user_timeout: self_pg_connection_pool_tcp_user_timeout,
+        } = self;
+        let Self {
+            pg_connection_pool_max_size: other_pg_connection_pool_max_size,
+            pg_connection_pool_max_wait: other_pg_connection_pool_max_wait,
+            pg_connection_pool_ttl: other_pg_connection_pool_ttl,
+            pg_connection_pool_ttl_stagger: other_pg_connection_pool_ttl_stagger,
+            pg_connection_pool_connect_timeout: other_pg_connection_pool_connect_timeout,
+            pg_connection_pool_tcp_user_timeout: other_pg_connection_pool_tcp_user_timeout,
+        } = other;
+        if let Some(v) = other_pg_connection_pool_max_size {
+            *self_pg_connection_pool_max_size = Some(v);
+        }
+        if let Some(v) = other_pg_connection_pool_max_wait {
+            *self_pg_connection_pool_max_wait = Some(v);
+        }
+        if let Some(v) = other_pg_connection_pool_ttl {
+            *self_pg_connection_pool_ttl = Some(v);
+        }
+        if let Some(v) = other_pg_connection_pool_ttl_stagger {
+            *self_pg_connection_pool_ttl_stagger = Some(v);
+        }
+        if let Some(v) = other_pg_connection_pool_connect_timeout {
+            *self_pg_connection_pool_connect_timeout = Some(v);
+        }
+        if let Some(v) = other_pg_connection_pool_tcp_user_timeout {
+            *self_pg_connection_pool_tcp_user_timeout = Some(v);
+        }
+    }
+
+    /// Applies the parameter values to persist's in-memory config object.
```

Comment:
nit: "persist" copy-pasta. here and elsewhere

Reply:
sorry about these!